### ISSUE TITLE
Attempting to call loginRedirect with a B2C-registered application fails due to unsupported scopes.

### ISSUE THREAD

**Issue by HorribleJibe** [AzureAD/microsoft-authentication-library-for-js#6355]
### Core Library

MSAL.js (@azure/msal-browser)

### Core Library Version

3.0.1

### Wrapper Library

Not Applicable

### Wrapper Library Version

None

### Public or Confidential Client?

Public

### Description

Hi team,

When attempting to call `loginRedirect` or `loginPopup` with a B2C application, `msal-browser` throws an error indicating that the request scopes are not supported. Attempting to overwrite either call with specific scopes also fails with the same message.

`invalid_request: AADB2C90012: The scope 'openid profile offline_access' provided in request is not supported.
Correlation ID: c99886fc-4ab6-48de-9cf3-5492086fdead`

This only occurs with msal-browser@3 - downgrading to msal-browser@2 allows login to occur as expected.

My understanding is that `profile` is a required scope for the library now, but B2C does not allow me to add this permission in the `API permissions` screen for any App Registration - only "openid" and "offline_access".

Am I doing something wrong?

### Error Message

ServerError: invalid_request: AADB2C90012: The scope 'openid offline_access profile' provided in request is not supported.
Correlation ID: 92ca825d-5572-46b6-9894-76413b005a81
Timestamp: 2023-08-17 00:11:34Z

    at _ResponseHandler.validateServerAuthorizationCodeResponse (ResponseHandler.ts:140:19)
    at AuthorizationCodeClient.handleFragmentResponse (AuthorizationCodeClient.ts:214:25)
    at RedirectHandler.handleCodeResponseFromHash (RedirectHandler.ts:171:48)
    at RedirectClient.handleHash (RedirectClient.ts:465:41)
(anonymous) @ index.ts:59
(anonymous) @ vue-router.mjs:3466
triggerError @ vue-router.mjs:3466
(anonymous) @ vue-router.mjs:3186
Promise.catch (async)
pushWithRedirect @ vue-router.mjs:3180
push @ vue-router.mjs:3112
install @ vue-router.mjs:3551
use @ runtime-core.esm-bundler.js:3752
(anonymous) @ main.ts:41
setTimeout (async)
(anonymous) @ __uno.css:20
(anonymous) @ __uno.css:20
Show 7 more frames
Show less

### MSAL Configuration

```javascript
export const msalConfig: Configuration = {
  auth: {
    clientId: import.meta.env.VITE_MSAL_CLIENT_ID, 
    authority: import.meta.env.VITE_MSAL_AUTHORITY,
    knownAuthorities: [import.meta.env.VITE_MSAL_KNOWN_AUTHORITY],
    redirectUri: import.meta.env.VITE_MSAL_REDIRECT_URI, 
    postLogoutRedirectUri: "/",
  }
};
```

### Relevant Code Snippets

```javascript
<script setup lang="ts">
import { msalInstance } from "~/msal/msal.instance";
const handleLogin = async () => {
  await msalInstance.loginRedirect();
};

handleLogin();
</script>

<template>
  <h1>Please wait</h1>
</template>
```

### Reproduction Steps

1. Navigate to page which calls msalInstance.loginRedirect() (or msalInstance.loginPopup()). Note: scopes passed explicitly are ignored/overwritten/merged.
2. SPA page redirects immediately to either the calling page (`/login`) or the redirectUri page in the config (`/authorise`, in my case).
3. Error message re: invalid scopes appears in console, one or more times.

### Expected Behavior

Login flow should occur as previously did.

### Identity Provider

Azure B2C Basic Policy

### Browsers Affected (Select all that apply)

Chrome, Firefox

### Regression

@azure/msal-browser@2

### Source

External (Customer)

**Comment by maxle0702** [comment]
got a same issue with this config:

export const msalConfig: Configuration = {
  auth: {
    clientId: process.env.CLIENT_ID,
    authority:
      `https://${process.env.DOMAIN_NAME}.b2clogin.com/${process.env.DOMAIN_NAME}.onmicrosoft.com/b2c_1_susi`,
    knownAuthorities: [`${process.env.DOMAIN_NAME}.b2clogin.com`],
    redirectUri: "http://localhost:5173/",
    postLogoutRedirectUri: "http://localhost:5173/",
    navigateToLoginRequestUrl: false,
  },
  cache: {
    cacheLocation: "sessionStorage",
    storeAuthStateInCookie: false,
  },
};

**Comment by enqueue** [comment]
see #6340

**Comment by Core121** [comment]
Wish we could get a quick release for this, we have yet to get a working release for B2C on v3.

**Comment by bilalahmed24** [comment]
Experiencing same issue. Had been waiting for this v3 stable version since months :/

**Comment by smj187** [comment]
i just downgraded to the following packages to get it working on a newly created vite app

> "@azure/msal-browser": "^2.38.0",
>  "@azure/msal-react": "^1.5.9",

**Comment by tnorling** [comment]
This has been patched in #6342 and will be included in the next release. In the meantime workarounds were shared in #6340
